src/preprocess.py

```diff
@@ -0,0 +1,154 @@
+import audio_parameters as ap
+import pandas as pd
+from torch.utils.data import Dataset, DataLoader
+import os
+import librosa
+import numpy as np
+from data import raw_text_to_phoneme_ids, data_path
+import collections
+from scipy import signal
+import torch as t
+import math
+
+
+class LJDatasets(Dataset):
+    """LJSpeech dataset."""
+
+    def __init__(self, csv_file, root_dir):
+        """
+        Args:
+            csv_file (string): Path to the csv file with annotations.
+            root_dir (string): Directory with all the wavs.
+
+        """
+        self.landmarks_frame = pd.read_csv(csv_file, sep='|', header=None)
+        self.root_dir = root_dir
+
+    def load_wav(self, filename):
+        return librosa.load(filename, sr=ap.sr)
+
+    def __len__(self):
+        return len(self.landmarks_frame)
+
+    def __getitem__(self, idx):
+        wav_name = os.path.join(self.root_dir, self.landmarks_frame.ix[idx, 0]) + '.wav'
+        original_text = self.landmarks_frame.ix[idx, 1]
+
+        text = np.asarray(raw_text_to_phoneme_ids(original_text), dtype=np.int32)
+        mel = np.load(wav_name[:-4] + '.pt.npy')
+        mel_input = np.concatenate([np.zeros([1,ap.num_mels], np.float32), mel[:-1,:]], axis=0)
+        text_length = len(text)
+        pos_text = np.arange(1, text_length + 1)
+        pos_mel = np.arange(1, mel.shape[0] + 1)
+
+        sample = {'text': text, 'mel': mel, 'text_length':text_length, 'mel_input':mel_input, 'pos_mel':pos_mel, 'pos_text':pos_text}
+
+        return sample
+
+class PostDatasets(Dataset):
+    """LJSpeech dataset."""
+
+    def __init__(self, csv_file, root_dir):
+        """
+        Args:
+            csv_file (string): Path to the csv file with annotations.
+            root_dir (string): Directory with all the wavs.
+
+        """
+        self.landmarks_frame = pd.read_csv(csv_file, sep='|', header=None)
+        self.root_dir = root_dir
+
+    def __len__(self):
+        return len(self.landmarks_frame)
+
+    def __getitem__(self, idx):
+        wav_name = os.path.join(self.root_dir, self.landmarks_frame.ix[idx, 0]) + '.wav'
+        mel = np.load(wav_name[:-4] + '.pt.npy')
+        mag = np.load(wav_name[:-4] + '.mag.npy')
+        sample = {'mel':mel, 'mag':mag}
+
+        return sample
```

Comment:
What's the purpose of the PostDataset? Is this another dataset or is this to be used with the test set?

Reply:
not needed, was used in the old repo, i just copied and didn't delete it